File: 2015/python/December-10/code.py
from tools.dl import read_data_from_serv
from tools.dl import read_data_from_file

# блок решения
def code_part1(lines:list[str]=None, irule:int=1, try_count: int=0):
    if lines is None:
        return
    if try_count == 0:
        return
    answer = 0
    for row in lines:
        if len(row) == 0:
            continue
        print('new line')
        row0 = row
        for itry in range(try_count):
            row1 = []
            pos = 0
            length = len(row0)
            while True:
                icount = 1
                letter = row0[pos]
                #print("ch letter ", letter, pos)
                if pos +1 == length:
                    row1.append(str(icount) + letter)
                    break
                else:
                    posinit = pos
                    for i in range(pos+1, length):
                        letter2 = row0[i]
                        #print("next:",letter2, pos)
                        if letter2 == letter:
                            icount+=1
                        else:
                            pos = i
                            row1.append(str(icount) + letter)
                            break
                    #print(pos, posinit, icount, length)
                    if posinit + icount == length:
                        row1.append(str(icount) + letter)
                        break
            row0 = ''.join(row1)
            print("try ", itry, ": ", len(row0))
        answer = len(row0)
    print("answer:", answer)


# входные данные
in_put = ['1', '11', '21', '1211', '111221']#,
in_put = ['1113122113']#read_data_from_serv(2015,10)
#in_put = read_data_from_file('in.txt')

# вычисление
code_part1(in_put, 1, 50)
#code_part1(in_put, 2, 50) #217 is low , 963 is high

File: 2015/cpp/December-10/code.cpp
#include <string>
#include <iostream>
#include <sstream>
#include <fstream>
using namespace std;
std::string work_on(std::string sIn)
{
	string res = "";
	std::stringstream row1;
	int pos = 0, length = sIn.length();
	while (1) {
		int icount = 1;
		string letter = sIn.substr(pos, 1);
		if (pos + 1 == length) {
			row1 << icount << letter;
			break;
		}
		else {
			int posinit = pos;
			for (int i = pos + 1; i < length; i++) {
				string letter2 = sIn.substr(i, 1);
				if (letter2 == letter)
					icount++;
				else {
					pos = i;
					row1 << icount << letter;
					break;
				}
			}
			if (posinit + icount == length) {
				row1 << icount << letter;
				break;
			}
		}
	}

	return row1.str();
}

int main()
{
	std::string sInput = "1113122113", sWork;
	for (int iTry=0; iTry < 50;iTry++)
	{
		sWork = work_on(sInput);
		sInput = sWork;

		cout << iTry <<"\t"<< sWork.length() << "\n";
	}
	cout << sWork.length() << "\n";
	system("pause");
    return 0;
}